prow/cmd/branchprotector/protect.go

```diff
@@ -276,11 +279,32 @@ func (p *protector) UpdateBranch(orgName, repo string, branchName string, branch
 		logrus.Infof("%s/%s=%s: already unprotected", orgName, repo, branchName)
 		return nil
 	}
+
+	// The github API currently does not support listing protections for all
+	// branches of a repository. We therefore have to make individual requests
+	// for each branch.
+	currentBP, err := p.client.GetBranchProtection(orgName, repo, branchName)
+	if err != nil {
+		return fmt.Errorf("get current branch protection: %v", err)
+	}
+	var policyBP github.BranchProtection
 	var req *github.BranchProtectionRequest
 	if *bp.Protect {
 		r := makeRequest(*bp)
 		req = &r
+		policyBP = github.BranchProtection{
+			EnforceAdmins:              r.EnforceAdmins != nil && *r.EnforceAdmins,
+			RequiredPullRequestReviews: r.RequiredPullRequestReviews,
+			RequiredStatusChecks:       r.RequiredStatusChecks,
+			Restrictions:               r.Restrictions,
+		}
 	}
+
+	if req != nil && equalBranchProtections(*currentBP, policyBP) {
+		logrus.Infof("%s/%s=%s: current branch protection matches policy, skipping", orgName, repo, branchName)
```

Comment:
Please change this to the debug level.
```suggestion
		logrus.Debugf("%s/%s=%s: current branch protection matches policy, skipping", orgName, repo, branchName)
```

Reply:
Thanks, will update the PR.